src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -913,7 +931,8 @@ public void MarkSymbolEndAnalysisComplete(ISymbol symbol, DiagnosticAnalyzer ana
             Debug.Assert(startActions.Any() || endActions.Any() || actions.Any());
             Debug.Assert(!executableBlocks.IsEmpty);
 
-            if (isGeneratedCode && _shouldSkipAnalysisOnGeneratedCode(analyzer))
+            if (isGeneratedCode && _shouldSkipAnalysisOnGeneratedCode(analyzer) ||
+                _isAnalyzerSuppressedForTree(analyzer, declaredNode.SyntaxTree))
```

Comment:
>_isAnalyzerSuppressedForTree(analyzer, declaredNode.SyntaxTree) [](start = 16, length = 63)

Do we need to check this here? Wouldn't this analyzer have been skipped in some parent context?

Reply:
No, there isn't any parent context which would have skipped this callback. The place where we skip these are just prior to the core analyzer callbacks (potentially few layers up the stack in case of some helpers around them). This ensures any core state tracking functionality is not affected by skipping callbacks. I have also made sure the logic is added to the same places where we skip generated code callbacks - this logic has been in place for few years and is well tested.